src/SFA.DAS.CaptureTheFlag.Web/Services/ForumService.cs

```diff
@@ -0,0 +1,75 @@
+﻿using DAS_Capture_The_Flag.Models.Forum;
+using Microsoft.AspNetCore.Identity;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using DAS_Capture_The_Flag.Data;
+using Microsoft.EntityFrameworkCore;
+
+namespace DAS_Capture_The_Flag.Services
+{
+    public class ForumService : IForumService
+    {
+        // Pass an instant of our dbContext
+        private readonly ApplicationDbContext _db;
+
+        public ForumService(ApplicationDbContext db)
+        {
+            _db = db;
+        }
+
+        public void Create(Forum forum)
+        {
+            _db.Forums.Add(forum);
+            _db.SaveChanges();
+        }
+
+        public void Delete(int forumId)
+        {
+            var topic = _db.Forums.Find(forumId);
+            _db.Forums.Remove(topic);
+        }
+
+        public IEnumerable<Forum> GetAll() // returns instance of forum from our database
+        {
+            return _db.Forums.Include(f => f.Posts);
+        }
+
+        //public IEnumerable<IdentityUser> GetAllActiveUsers()
+        public IEnumerable<ApplicationUser> GetAllActiveUsers()
+        {
+            throw new NotImplementedException();
+        }
+
+        public Forum GetById(int id)
```

Comment:
Future improvement - There could be a lot of posts in the forum so a paging strategy for the posts could help
See the .Skip and .Take methods